crates/ra_proc_macro/src/process.rs

```diff
@@ -0,0 +1,212 @@
+//! Handle process life-time and message passing for proc-macro client
+
+use crossbeam_channel::{bounded, Receiver, Sender};
+use ra_tt::Subtree;
+
+use crate::msg::{ErrorCode, Message, Request, Response, ResponseError};
+use crate::rpc::{ExpansionResult, ExpansionTask, ListMacrosResult, ListMacrosTask, ProcMacroKind};
+
+use io::{BufRead, BufReader};
+use std::{
+    convert::{TryFrom, TryInto},
+    io::{self, Write},
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+    thread::{spawn, JoinHandle},
+};
+
+#[derive(Debug, Default)]
+pub(crate) struct ProcMacroProcessSrv {
+    inner: Option<Sender<Task>>,
+}
+
+#[derive(Debug)]
+pub(crate) struct ProcMacroProcessThread {
+    handle: Option<JoinHandle<()>>,
```

Comment:
Let's use `jod_thread::JoinHandle` here, to make sure stuff is properly cleaned up:

https://github.com/rust-analyzer/rust-analyzer/pull/3754

Reply:
Fixed.